src/expand.cpp

```diff
@@ -86,6 +87,7 @@ namespace Sass {
 
   Statement* Expand::operator()(Ruleset* r)
   {
+    bool old_at_root_without_rule = this->at_root_without_rule;
```

Comment:
I case you need better control you could create a variable scope local enough for your use-case.